frontend/packages/kubevirt-plugin/src/models.ts

```diff
@@ -0,0 +1,136 @@
+import { K8sKind } from '@console/internal/module/k8s';
+
+export const VirtualMachineModel: K8sKind = {
```

Comment:
@mareklibra @spadgett All KubeVirt related models are CRDs, right?

Shouldn't we also add `crd: true` to these model definitions?

Reply:
Right, should be.

But without custom router rule (would require additional extension), the URL is not user-friendly (see `referenceForModel()`, contains king and api).

In addition, we register plugin models under `allModels()`, logic around `connectToPlural()` would need to be improved for that.

I am more for keeping recent state (without `crd: true`), especially due to nicer URL without additional touching router.

Reply:
Requirements to consider when making decision here:
- `kubevirt` is expected to be first-class citizen in OpenShift, the user expects the URL to be as pretty as for ie. `Pod`
- as a rule, there can be just a single `kubevirt` deployment in a cluster at a time. It means, there can be just a single `version` of our CRDs at a time.
- we can improve URL via extending `Router`, but considering ^^^ I don't think it is neccessary.

Reply:
cc @spadgett we would like to hear your opinion on this.

Not using `crd: true` would make KubeVirt resource URLs shorter/prettier but also imply that Console sees such resources as base k8s objects (I'm not saying this is a bad thing, just saying that they are technically custom resources managed by KubeVirt operator).

Using `crd: true` plus a React router alias/redirect (added via extension system) for shorter/prettier URLs is another possibility.

Reply:
I would suggest `crd: true` for now, although I think we should get rid of the `crd` property in the model object if we can. It's a bit misleading since not all CRDs have it, and some resources have it that aren't CRDs. We used to key on it for other behavior changes around watches, but I think we've removed that.

Honestly, I don't think most users will notice or care if the URL doesn't use the short name. The biggest problem is that we include the API version in the URL, not just the group. This could break links if versions change (for instance, v1alpha1 -> v1). But that's outside the scope of this PR. We need to look at this problem generally, not just for KubeVirt.